### DISCUSSION TITLE
corrupt stream with rtsp-simple-server - but direct rtsp to camera works perfect

### DISCUSSION THREAD

**Discussion by itakka01** [bluenviron/mediamtx#850] [Questions] [answered]
Hello, 

I have troubles with my TAPO C_100 camera. Getting the stream directly from the camera (for example with vlc or cctv-viewer) works perfect. But when trying to get the stream over rtps-simple-server it is corrupt or sometimes smearing like the hell.
--> cut rtsp-simple-server.yml
  tapo_c100_01:
    source: rtsp://user:password@192.168.2.120/stream2
    sourceProtocol: udp
--> endcut
any idea how to overcome this problem?

![tapo_c_100_direct](https://user-images.githubusercontent.com/48524599/154863410-55c5bc15-d296-4f37-907d-e68191bb6dfa.jpg)
![tapo_c_100_rtsp-simple-streamer](https://user-images.githubusercontent.com/48524599/154863417-cac656ad-ac95-44a3-8bc5-9a9994f7b9aa.jpg)
![tapo_c_100_direct](https://user-images.githubusercontent.com/48524599/154863421-74b5a84a-1957-4561-9be9-396337c57992.jpg)
![tapo_c_100_rtsp-simple-streamer](https://user-images.githubusercontent.com/48524599/154863425-a773edec-7724-4271-af76-0c8f98d6b024.jpg)

**Comment by aler9** [comment, marked as answer]
Please read here:

https://github.com/aler9/rtsp-simple-server#corrupted-frames

**Reply by itakka01** [reply]
thank you!  that solved my problem. I used the following parameters for the TAPCO_C100 cams: 

[...]
readBufferCount: 512
readBufferSize:  10240
tapo_c100_01:
    source: rtsp://<user>:<password>@192.168.2.120/stream2
    sourceProtocol: udp
[...]

regards
Bernhard

[a comment by an automated account is not shown]
